app/containers/SearchTrackContainer/index.js

```diff
@@ -0,0 +1,141 @@
+/**
+ *
+ * SearchTrackContainer Container
+ *
+ */
+
+import React, { useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { injectIntl } from 'react-intl';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import { injectSaga } from 'redux-injectors';
+import { Input, Card } from 'antd';
+import { selectSearchTrackName, selectTracksData, selectTracksError } from './selectors';
+import { debounce, get, isEmpty, isNull, isUndefined } from 'lodash';
+import { searchTrackContainerCreators } from './reducer';
+import searchTrackContainerSaga from './saga';
+import If from '@app/components/If/index';
+import For from '@app/components/For/index';
+import TracksCard from '@app/components/TracksCard/index';
+import styled from 'styled-components';
+
+const MainBox = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const Container = styled.div`
+  && {
+    display: grid;
+    grid-template-columns: repeat(4, 1fr);
+    width: 100%;
+    margin: 0 auto;
+  }
+`;
+
+const TrackCustomCard = styled(Card)`
+  && {
+    width: 100%;
+  }
+`;
+
+const { Search } = Input;
+
+export function SearchTrackContainer({
+  dispatchTracksResult,
+  dispatchClearTracksResult,
+  tracksData,
+  tracksError,
+  searchedTrackName
+}) {
+  useEffect(() => {
+    if (searchedTrackName && !tracksData?.items?.length) {
+      dispatchTracksResult(searchedTrackName);
+    }
+    return () => {
+      dispatchClearTracksResult();
+    };
+  }, []);
+
+  const handleTrackSearch = (trackName) => {
+    if (trackName && trackName !== undefined) {
```

Comment:
if (trackName) is sufficient. Only if its present it will get into the condition otherwise it will get into the else block